beluga_tutorial/src/main.cpp

```diff
@@ -0,0 +1,333 @@
+   */
+  double sensor_model_sigam{1.0};
+
+  /// Minimum particle weight.
+  /**
+   * Used to keep all particles "alive" in the reweight step.
+   */
+  double min_particle_weight{0.08};
+
+  /// Dataset path.
+  /**
+   * The dataset file is used to save the data produced by the simulation for posterior analisys.
+   */
+  std::filesystem::path dataset_path{"./src/beluga/beluga_tutorial/bags/dataset.yaml"};
+
+  // Name of the landmark_map to save or load from.
+  std::string landmark_map_file_name{"landmark_map.yaml"};
+};
+
+using Particle = std::tuple<double, beluga::Weight>;
+using LandmarkMapVector = std::vector<double>;
+using SensorData = std::vector<double>;
+using GroundTruth = double;
+using Estimate = std::pair<double, double>;
+
+struct TutorialData {
+  GroundTruth ground_truth;
+  beluga::TupleVector<Particle> current;
+  beluga::TupleVector<Particle> propagate;
+  beluga::TupleVector<Particle> reweight;
+  beluga::TupleVector<Particle> resample;
+  Estimate estimation;
+};
+
+struct TutorialDataset {
+  LandmarkMapVector landmark_map;
+  std::vector<TutorialData> sim_data;
+};
+
+void load_params_from_yaml(const std::filesystem::path& path, TutorialParams& tutorial_params) {
+  try {
+    YAML::Node params = YAML::LoadFile(path);
+    tutorial_params.load_landmark_map = params["load_landmark_map"].as<bool>();
+    tutorial_params.save_landmark_map = params["save_landmark_map"].as<bool>();
+    tutorial_params.map_size = params["map_size"].as<int>();
+    tutorial_params.number_of_doors = params["number_of_doors"].as<int>();
+    tutorial_params.number_of_particles = params["number_of_particles"].as<int>();
+    tutorial_params.number_of_cycles = params["number_of_cycles"].as<int>();
+    tutorial_params.initial_pose = params["initial_pose"].as<double>();
+    tutorial_params.initial_pose_sd = params["initial_pose_sd"].as<double>();
+    tutorial_params.dt = params["dt"].as<double>();
+    tutorial_params.velocity = params["velocity"].as<double>();
+    tutorial_params.translation_sd = params["translation_sd"].as<double>();
+    tutorial_params.sensor_range = params["sensor_range"].as<double>();
+    tutorial_params.sensor_model_sigam = params["sensor_model_sigam"].as<double>();
+    tutorial_params.min_particle_weight = params["min_particle_weight"].as<double>();
+    tutorial_params.dataset_path = params["dataset_path"].as<std::string>();
+    tutorial_params.landmark_map_file_name = params["landmark_map_file_name"].as<std::string>();
+  } catch (YAML::BadFile& e) {
+    std::cout << e.what() << "\n";
+  }
+}
+
+void load_landmark_map_from_yaml(const std::filesystem::path& path, LandmarkMapVector& landmark_map) {
+  try {
+    YAML::Node node = YAML::LoadFile(path);
+    landmark_map = node["landmark_map"].as<LandmarkMapVector>();
+  } catch (YAML::BadFile& e) {
+    std::cout << e.what() << "\n";
+  }
+}
+
+void save_landmark_map_to_yaml(const std::filesystem::path& path, LandmarkMapVector& landmark_map) {
```

Comment:
In the spirit of keeping this as simple as possible. Do we need this functionality?

Reply:
This option was added in case you want to run the algorithm again on the same map, without the need to copy the map manually to a file. I agree that it adds more functions to the tutorial. I added it to discuss it with you to see what you think.

Reply:
I think the function to load a map from a file already accomplishes this. I would also remove the random map generation feature as having a static map seems to be sufficient for most tests.